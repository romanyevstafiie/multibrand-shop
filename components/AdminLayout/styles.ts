import styled from 'styled-components';

export const LayoutContainer = styled.section`
  display: grid;
  grid-template-columns: 1fr 8fr;
  min-height: 100vh;
`;

export const ChildrenWrapper = styled.main`
  background-color: ${({ theme }) => theme.dashboardGrayBg};
`;
