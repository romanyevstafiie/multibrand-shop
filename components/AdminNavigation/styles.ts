import styled from 'styled-components';

export const StyledAdminNavigation = styled.aside`
  display: flex;
  align-items: center;
  justify-content: center;
  position: sticky;
  top: 0;
  padding: 5rem 1rem;

  li {
    margin: 1rem 0;
  }
`;
