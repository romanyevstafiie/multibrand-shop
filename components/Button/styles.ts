import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
