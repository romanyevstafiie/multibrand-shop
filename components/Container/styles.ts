import styled from 'styled-components';

export const StyledContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
  padding: 0 1rem;

  a {
    text-decoration: none;
  }
`;
