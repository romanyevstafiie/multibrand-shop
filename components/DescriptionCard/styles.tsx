import styled from 'styled-components';

export const CardContainer = styled.article`
  background: ${({ theme }) => theme.secondaryBg};
  border-radius: ${({ theme }) => theme.borderRadius};
  border: 1px solid ${({ theme }) => theme.borderGrayColor};
  padding: 1.25rem;
`;
export const Description = styled.p``;
