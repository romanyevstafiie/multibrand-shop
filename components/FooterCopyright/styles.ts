import styled from 'styled-components';

export const FooterCopyrightSection = styled.section`
  font-size: 0.75rem;
  color: ${({ theme }) => theme.lightGray};

  ul {
    display: flex;
    li:nth-child(n + 2) {
      margin-left: 0.75rem;
    }
  }
  .router-link {
    color: ${({ theme }) => theme.lightGray};
  }
`;
