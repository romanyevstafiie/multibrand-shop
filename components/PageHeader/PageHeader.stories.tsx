import { Story } from '@storybook/react';
import React from 'react';
import PageHeader, { Props } from './PageHeader';

export default {
  title: 'Typography/PageHeader',
  component: PageHeader,
};

const Template: Story<Props> = (args) => <PageHeader {...args} />;

export const Primary = Template.bind({});
Primary.args = { headerText: 'Add new product' };
