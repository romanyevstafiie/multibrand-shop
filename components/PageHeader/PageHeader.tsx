import React from 'react';
import styled from 'styled-components';

export interface Props {
  headerText: string;
}

/**
 * Page <h1> Header
 *@function PageHeader
 *@param {string} headerText - Header text content
 *@returns {JSX.Element} - Rendered component (or null if `success` prop is failed)
 */
const PageHeader = ({ headerText }: Props): JSX.Element => {
  return <StyledHeader>{headerText}</StyledHeader>;
};

const StyledHeader = styled.h2`
  font-size: 2rem;
  text-transform: capitalize;
  font-weight: 700;
  margin: 1rem 0;
  @media ${({ theme }) => theme.media.tablet} {
    font-size: 1.25rem;
  }
`;

export default PageHeader;
