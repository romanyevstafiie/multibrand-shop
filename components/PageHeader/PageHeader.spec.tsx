import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import PageHeader from './PageHeader';
import { withReduxAndStyledProviders } from '../../test/testUtils';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from '../../styles/mainTheme';

/**
 * Setup function for the component
 * @returns {ShallowWrapper}
 */
const setup = () => {
  return mount(withReduxAndStyledProviders(<PageHeader headerText="tomato" />));
};

describe('Page Header', () => {
  const wrap = setup();

  it('should render text content without throwing an error', () => {
    expect(wrap);
    expect(wrap.find('h2').text()).toBe('tomato');
  });
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={mainTheme}>
          <PageHeader headerText="tomato" />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
