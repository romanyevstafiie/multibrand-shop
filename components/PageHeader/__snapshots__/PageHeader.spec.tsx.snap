// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
.c0 {
  font-size: 2rem;
  text-transform: capitalize;
  font-weight: 700;
  margin: 1rem 0;
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.25rem;
  }
}

<h2
  className="c0"
>
  tomato
</h2>
`;
